src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -921,11 +925,11 @@ public static void SerializeInner(object obj, BinaryTokenStreamWriter stream, Ty
                 return;
             }
 
-            var t = obj.GetType();
+            var typeInfo = obj.GetType().GetTypeInfo();
```

Comment:
This place scares me most! It is very hot path, called for every argument in every grain call. Do we need both GetType().GetTypeInfo();?